xarray_array_testing/tests/test_cubed.py

```diff
@@ -0,0 +1,83 @@
+from contextlib import nullcontext
+from typing import ContextManager
+
+import cubed
+import cubed.random
+import hypothesis.strategies as st
+import numpy as np
+import numpy.testing as npt
+import pytest
+from hypothesis import note
+
+from xarray_array_testing.base import DuckArrayTestMixin
+from xarray_array_testing.creation import CreationTests
+from xarray_array_testing.reduction import ReductionTests
+
+
+def cubed_random_array(shape: tuple[int], dtype: np.dtype) -> cubed.Array:
+    """
+    Generates a random cubed array
+
+    Supports integer and float dtypes.
+    """
+    # TODO hypothesis doesn't like us using random inside strategies
+    rng = np.random.default_rng()
```

Comment:
Consider using a Hypothesis-provided seed?  I'd also be happy to accept a PR to generate Numpy prng instances 🙂

Reply:
we should probably try using `xps.arrays()` instead

Reply:
The other argument against is that sometimes you just want a faster PRNG for the elements; the distribution is a bit less likely to find bugs but setting elements individually is a _lot_ slower (even though we do a sparse subset)